cmake/dependencies.cmake

```diff
@@ -7,6 +7,7 @@ set(ENV{CPM_SOURCE_CACHE} "${PROJECT_SOURCE_DIR}/.cpmcache")
 
 include(${PROJECT_SOURCE_DIR}/cmake/fetch_boost.cmake)
 
+fetch_boost_library(core)
```

Comment:
@blozano-tt @TT-billteng @afuller-TT  I thought our position was to move away from boost? What do we think?

Reply:
Its always nice to depend on the standard library, and incur fewer dependencies.

@patrickroberts can you expand on why we want to add boost for Span?

Also, it looks like it gets linked even if it doesn't get used based on the fallback scheme?

Reply:
I, too, would like to hear why Boost is needed before we incur the cost of it (esp. in so many places).

Reply:
I'll remove it, we already are using `<span>` other places in the library. I had gone this route because I was instructed not to rely on C++20 features when I made the suggestion for this API improvement. If we need C++17 compatibility, we can revisit this topic at that point.

Reply:
Based on @ayerofieiev-tt's comments on other PRs I thought we needed C++17 to be compatible for automotive use

Reply:
After discussion yesterday and confirmation today (with management and customers), I've left the feature detection in place, and conditionally `fetch_boost_library(core)` if `CMAKE_CXX_STANDARD LESS 20` to polyfill `std::span` with `boost::span`. A long-term goal is to achieve C++17 compatibility, and it doesn't make sense to hand-roll a container that already has a perfectly acceptable open source implementation already available for use. Boost is not exposed in the API and is used as an implementation detail for this, and the expectation is for the rest of tt-metal to use `tt::stl::Span` instead of `std::span`.